### ISSUE TITLE
Support Timestamp extension type

### ISSUE THREAD

**Issue by jm-** [knopp/msgpack_dart#12]
It would be great if the [timestamp extension type](https://github.com/msgpack/msgpack/blob/master/spec.md#timestamp-extension-type) is supported out of the box.

It seems like there are already [some functions that test DateTime deserialization](https://github.com/knopp/msgpack_dart/blob/master/test/msgpack_dart_test.dart#L504), but these aren't wired up to the main test entrypoint. I have tried running these locally and the serialized DateTime currently deserializes to `<null>`.

```
$ dart test
Building package executable...
Built test:test.
00:00 +0 -1: test\messagepack_test.dart: unpackSmallDateTime [E]
  Expected: DateTime:<1970-01-01 13:00:00.000>
    Actual: <null>

  package:test_api                 expect
  test\messagepack_test.dart 14:3  unpackSmallDateTime
```

**Comment by AntonEriksson978** [comment]
Yes, I also have this problem. This msgpack implementation doesn't seem to be able to parse datetime.
